src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -464,6 +464,16 @@ public AuthenticationConfiguration useKeyManagerCredential(X509KeyManager keyMan
     }
 
     /**
+     * Create a new configuration which is the same as this configuration, but which uses the given choice.
+     *
+     * @param choice the choice to use
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration useChoice(String choice) {
```

Comment:
The weakness of having a blanket use-choice configuration is that the choice callback is completely open-ended... there's no way to establish what kind of choice you're making.  It might be better to cause it to match information about the choice, e.g. also accept a ``Predicate<String>`` which tests the prompt string, or a ``BiPredicate<Class<? extends ChoiceCallback>, String>`` which checks both the callback type _and_ the prompt string and then only respond in this case.  This would also solve the problems that... (see below)